### DISCUSSION TITLE
GUI Enhancement

### DISCUSSION THREAD

**Discussion by LMBooth** [pieeg-club/PiEEG#3] [Ideas]
Hey, i don't have a pieeg system but i had a look at the plotting code for the gui and thought i'd make a discussion point, 
i made something similar for a eeg system and thought i'd share my findings. For live streaming data in to a plotter i'd recommend using pyqtgraph in a pyqt gui, where data is received in a thread and passed via signal to the gui pyqtgraph instead of using matplotlib. Pyqtgraph can update so much faster making it much better suited to real time streaming and means you'd have a non blocking gui. If this discussion is unnecessary please do close.

It would also be great if one could optionally turn on and setup a pyLSL outlet too, so the data could be pushed out to multiple other sources making for better integration in to other modules and systems.

**Comment by Ildaron** [comment]
Hey! Yes, I need to try it, thank you so much!
